AI #2359: Add Actor (`Consumer ID` and `Principal ID`) columns to `Cost and Usage` dataset (#2360)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/glossary.md

A service-provider offering that allows a customer to enter into a [*commitment*](#glossary:commitment). Commitment programs include [*commitment discounts*](#glossary:commitment-discount), [*capacity reservations*](#glossary:capacity-reservation), and other constructs that require advance spend or usage agreements.

<a name="glossary:consumer"><b>Consumer</b></a>

> An authenticated actor (e.g., end-user, tenant, application, autonomous agent) that utilizes a [*resource*](#glossary:resource) or [*service*](#glossary:service). This is an application-layer actor benefiting from the usage, which may be distinct from the infrastructure-layer [*principal*](#glossary:principal) that initiated the request.

Reviewer: ```suggestion
A downstream actor (e.g., end-user, application, autonomous agent) that consumes [*resources*](#glossary:resource) or [*services*](#glossary:service). The consumer may be distinct from the [*principal*](#glossary:principal) that initiates requests for *resources* or *services*.
```

Author: I want the start of this definition to read the same in both Principal and Consumer.  How about for both, we just use "An authenticated actor"?

Reviewer: As mentioned [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/2360#issuecomment-4443659690) (see the 3rd assumption), I believe the Principal must be authenticated (identity verified), while the Consumer does not necessarily require authentication in all scenarios.

A comprehensive list of prices offered by a service provider.

<a name="glossary:principal"><b>Principal</b></a>

> An authenticated actor (e.g., user, service account, application credential) that directly initiates a request to a [*service provider*](#glossary:service-provider) to use or manage a [*resource*](#glossary:resource) or [*service*](#glossary:service). It represents the infrastructure-layer actor that authorized a charge, which may be distinct from the downstream [*consumer*](#glossary:consumer) that utilizes the *resource* or *service*.

Reviewer: ```suggestion
An authorized actor (e.g., user account, service account, application) that initiates requests for [*resources*](#glossary:resource) or [*services*](#glossary:service). The principal may be distinct from the downstream [*consumer*](#glossary:consumer) that consumes *resources* or *services*.
```

Author: "Authenticated and authorized" feels redundant.  Why both?  Would rather just use one or the other.

Reviewer: As mentioned [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/2360#issuecomment-4443659690) (see the 3rd assumption), I believe:

- The Principal must be authenticated (identity verified), while the Consumer does not necessarily require authentication in all scenarios.
- The Principal should be described as authorized (permitted to act), rather than as authorizing or as the actor that authorized a charge.

Whether authorization implies authentication is a separate question.

Author: I would be more than happy to use "authenticated" alone in place of "authorized" (or vice versa), if you're comfortable with that.

A comprehensive list of prices offered by a service provider.

<a name="glossary:principal"><b>Principal</b></a>

> An authorized actor (e.g., user account, service account, application) that initiates requests for [*resources*](#glossary:resource) or [*services*](#glossary:service).

Reviewer 2: Tightens the `Principal` glossary definition so it can't be misread as the end-user, per the TF-2 request in AI #2483 (distinguish the principal from the downstream consumer/end-user, e.g. a gateway or proxy role versus the end-user behind it).

`consumer` is intentionally plain text here rather than a glossary link, because the `Consumer` term is not defined until the ConsumerId carve-out (#2495). This matches how the `Operating Model` definition refers to "host providers" as plain text. When #2495 lands, this can be upgraded to a `[*consumer*](#glossary:consumer)` link.

```suggestion
The initiating or authorizing actor requesting [*resources*](#glossary:resource) or [*services*](#glossary:service). The Principal can differ from the downstream consumer or end-user on whose behalf the request is made (e.g., a gateway or proxy role acting for many end-users behind it).
```

> The initiating or authorizing actor requesting [*resources*](#glossary:resource) or [*services*](#glossary:service). The Principal can differ from the downstream consumer or end-user on whose behalf the request is made (e.g., a gateway or proxy role acting for many end-users behind it).

Reviewer: a principal is the entity that has been granted authorization to request resources or services (an authorized actor) — not an entity that grants authorization to others.

```suggestion
The initiating or authorized actor requesting [*resources*](#glossary:resource) or [*services*](#glossary:service). The Principal can differ from the downstream consumer or end-user on whose behalf the request is made (e.g., a gateway or proxy role acting for many end-users behind it).
```

Reviewer: ```suggestion
The actor that initiates or is authorized to request [*resources*](#glossary:resource) or [*services*](#glossary:service). The Principal can differ from the downstream consumer or end-user on whose behalf the request is made (e.g., a gateway or proxy role acting for many end-users behind it).
```

> An entity defined in a service provider identity and access management model (e.g., user, role, service account) to which access to [*resources*](#glossary:resource) or [*services*](#glossary:service) is granted. A principal is distinct from the credential (e.g., API key, access token) presented with an individual request, and from the organizational container (e.g., project, workspace) within which the request runs. The same principal may be associated with multiple credentials.

Reviewer 3: ```suggestion
An entity defined in an identity and access management model (e.g., user, role, service account) to which access to [*resources*](#glossary:resource) or [*services*](#glossary:service) is granted. A principal is distinct from the credential (e.g., API key, access token) presented with an individual request, and from the organizational container (e.g., project, workspace) within which the request runs. The same principal may be associated with multiple credentials.
```

A comprehensive list of prices offered by a service provider.

<a name="glossary:principal"><b>Principal</b></a>

An entity defined in an identity and access management model (e.g., user, role, service account) to which access to [*resources*](#glossary:resource) or [*services*](#glossary:service) is granted. A principal is distinct from the credential (e.g., API key, access token) presented with an individual request, and from the organizational container (e.g., project, workspace) within which the request runs. The same principal may be associated with multiple credentials.

<a name="glossary:requester"><b>Requester</b></a>

> The entity on whose behalf a request that produces a [*charge*](#glossary:charge) is made, as represented in an identity and access management model. A service provider commonly represents a requester at more than one level, such as the [*principal*](#glossary:principal) to which access to [*resources*](#glossary:resource) or [*services*](#glossary:service) is granted and the credential presented on the request. A requester is distinct from the party that benefits from the usage or purchase, which a FOCUS dataset does not identify.

Reviewer 4: ### F-005 — error — [Line 277](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/blob/71db3c5d345548586e7f70ef7301134fb8cd905f/specification/glossary.md#L277)

**Changed content:** `A service provider commonly represents a requester...`

**Rule:** `editorial-conventions-13` — Writing Specification Content & Review Guidelines > Editorial Conventions > Linking Rule; 13th indexed rule. “Link the first remaining occurrence in reading order.”

**Engineering context:** First-occurrence linking is evaluated independently within each glossary entry.

**Problem:** The first `service provider` reference in the Requester entry is unlinked.

**Impact:** The definition omits required navigation to a term on which its meaning depends.

**Behavioral effect of the correction:** No schema or normative behavior change.

**Suggested correction:** Link the first occurrence to the Service Provider glossary entry.

```suggestion
The entity on whose behalf a request that produces a [*charge*](#glossary:charge) is made, as represented in an identity and access management model. A [*service provider*](#glossary:service-provider) commonly represents a requester at more than one level, such as the [*principal*](#glossary:principal) to which access to [*resources*](#glossary:resource) or [*services*](#glossary:service) is granted and the credential presented on the request. A requester is distinct from the party that benefits from the usage or purchase, which a FOCUS dataset does not identify.
```

Reviewer 2: Agreed that `service provider` should be linked there, and it is better fixed in one pass than here. The glossary is not consistent on this today: 8 entries reference the term as plain text, 2 link it (Native Dataset, Operating Model) and 6 do not (Contract, Credit, Price List, Service, Tag Scheme, Virtual Currency). Linking it in Requester alone moves the count to 3 of 9 and leaves the same violation in the other six entries.

A single pass that fixes first-occurrence linking for `service provider` across all of them is consistency review work, where it lands as one reviewable change against one rule.

The same rule fires on the `service provider` reference in the PrincipalId uniqueness bullet in `principalid.md`, which is better left to that pass as well.

Reviewer 2: Correcting the record on this one: the batch-accept in 8d9d9033 took this suggestion in alongside my counter-suggestion on the column file, after I had declined it here. I restored the entry to plain text in ff664c1c so the branch matches that note, and the glossary-wide linking pass can take all nine service provider mentions together, including the six entries that are still plain.
